src/createServer.js

```diff
@@ -1,8 +1,51 @@
 'use strict';
 
+const http = require('http');
+const fs = require('fs');
+const path = require('path');
+
+const FILE_PATH = '/file';
+const PUBLIC_PATH = '/public';
+
 function createServer() {
-  /* Write your code here */
-  // Return instance of http.Server class
+  const server = http.createServer((req, res) => {
+    const { pathname } = new URL(req.url, `http://${req.headers.host}`);
+
+    res.setHeader('Content-Type', 'text/plain');
+
+    if (!pathname.includes(FILE_PATH)) {
+      res.statusCode = 400;
+      res.end('Hint: for load file `pathname` must start with `/file/`');
+
+      return;
+    }
+
+    if (pathname.includes('//')) {
+      res.statusCode = 404;
+      res.end('Double "//" not supported.');
+
+      return;
+    }
+
+    let publicPath = pathname.replace(FILE_PATH, PUBLIC_PATH);
+
+    if (publicPath === PUBLIC_PATH || publicPath === `${PUBLIC_PATH}/`) {
```

Comment:
the process of building `publicPath` is a bit enigmatic and can be simplified by doing something like this:
```javascript
const fileName = pathname.slice(6) || 'index.html';
const filePath = `./public/${fileName}`;
```
A disadvantage is that `6` is a magic number (it can be extracted to a variable)

Reply:
@Centmsn  I didn’t like the option where I guess the number of letters, so I did it exactly for every options
Changed it?

Reply:
I wouldn't call it a guessing as long as there is only one pattern allowed, which is `/file/`. As long as it stays this way, we can be sure that number `6` is correct. There are situations where using a fixed number is more readable solution (it's one of these situations in my opinion).

Reply:
I can't do it your way because we can have '/file' = 5 OR '/file/'=6
test failed
so I leave it